app/src/views/GrantRoundDetails.vue

```diff
@@ -0,0 +1,254 @@
+    </div>
+    <div v-else-if="grantRound.error">
+      <span>{{ grantRound.error }}</span>
+    </div>
+    <div v-else>
+      <span>Loading details...</span>
+    </div>
+  </div>
+
+  <!-- Contributing to GrantRound -->
+  <div v-else class="flex flex-col justify-center sm:px-6 lg:px-8">
+    <div class="sm:mx-auto sm:w-full sm:max-w-md">
+      <img
+        class="mx-auto h-12 w-auto"
+        src="https://tailwindui.com/img/logos/workflow-mark-indigo-600.svg"
+        alt="Workflow"
+      />
+      <h2 class="mt-6 text-center text-3xl font-extrabold text-gray-900">
+        Contribute to GrantRound {{ formatAddress(grantRound.address.toString()) }}
+      </h2>
+    </div>
+
+    <div class="mt-8 sm:mx-auto sm:w-full sm:max-w-md text-left">
+      <div class="py-8 px-4 border border-gray-200 shadow sm:rounded-lg sm:px-6 bg-gray-50">
+        <form class="space-y-6" @submit.prevent="sendContribution">
+          <!-- token address -->
+          <BaseInput
+            v-model="form.token"
+            description="The token used to make the contribution"
+            id="contribution-token-address"
+            label="Contribution Token address"
+            :readonly="true"
+            :disabled="true"
+            :rules="isValidAddress"
+            errorMsg="Please enter a valid address"
+          />
+
+          <!-- contribution amount -->
+          <BaseInput
+            v-model="form.amount"
+            description="The number of tokens to contribute"
+            id="contribution-amount"
+            label="Contribution amount"
+            :rules="(amount) => amount || 0 > 0"
+            errorMsg="Please enter an amount greater than 0"
+          />
+
+          <!-- Submit and cancel buttons -->
+          <button
+            type="submit"
+            class="btn btn-primary w-full"
+            :class="{ 'btn-primary-disabled': !isFormValid }"
+            :disabled="!isFormValid"
+          >
+            Send Contribution
+          </button>
+          <button @click.prevent="cancelContribution" class="btn btn-outline w-full">Cancel</button>
+        </form>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, ref } from 'vue';
+import { useRoute } from 'vue-router';
+import BaseInput from 'src/components/BaseInput.vue';
+// --- Store ---
+import useDataStore from 'src/store/data';
+import useWalletStore from 'src/store/wallet';
+// --- Methods and Data ---
+import { GRANT_ROUND_ABI, ERC20_ABI } from 'src/utils/constants';
+import { BigNumber, Contract, ContractTransaction } from 'src/utils/ethers';
+import { daysAgo, formatAddress, isValidAddress, isValidUrl } from 'src/utils/utils';
+// --- Types ---
+import { GrantRound } from '@dgrants/types';
+import { GrantRound as GrantRoundContract } from '@dgrants/contracts';
+import { BigNumberish, utils } from 'ethers';
```

Comment:
Can we get these from the `src/utils/ethers` file also? That way all ethers imports are in one place (might need to add new exports to that file, and only export the specific utils we need to keep bundle size down)

Reply:
Resolved in f095fea